include/pybind11/embed.h

```diff
@@ -189,6 +189,7 @@ class scoped_interpreter {
     scoped_interpreter &operator=(const scoped_interpreter &) = delete;
     scoped_interpreter &operator=(scoped_interpreter &&) = delete;
 
+    // NOLINTNEXTLINE(bugprone-exception-escape)
```

Comment:
If we caught the exception and explicitly call std::terminate that would also work well.